### PR TITLE
Checksum address in mosaic config

### PR DESCRIPTION
**Pull request by gulshanvasnani**
PR contains changes to write checksum address to mosaic-config.

fixes https://github.com/openst/facilitator/issues/162

### CODE REVIEW

File: src/NewChain/Initialization.ts [hunk 1]
```diff
@@ -180,14 +180,14 @@ export default class Initialization {
...
 
-    auxiliaryContracts.anchorOrganizationAddress = anchorOrganization.address;
-    auxiliaryContracts.anchorAddress = anchor.address;
-    auxiliaryContracts.ostCoGatewayOrganizationAddress = coGatewayAndOstPrimeOrganization.address;
-    auxiliaryContracts.ostPrimeAddress = ostPrime.address;
-    auxiliaryContracts.ostEIP20CogatewayAddress = ostCoGateway.address;
-    auxiliaryContracts.gatewayLibAddress = gatewayLib.address;
-    auxiliaryContracts.messageBusAddress = messageBus.address;
-    auxiliaryContracts.merklePatriciaLibAddress = merklePatriciaProof.address;
+    auxiliaryContracts.anchorOrganizationAddress = Web3.utils.toChecksumAddress(anchorOrganization.address);
```
Comment (puneet-khushwani-eth) [inline src/NewChain/Initialization.ts]: I feel that all calls to external methods should be wrapped in a method in our code. 

In this context, we are directly calling  `Web3.utils.toChecksumAddress` on `import Web3 = require('web3')`. If something inside `Web3` changes we would need to change the usages at a lot of places in our code base. 

I would propose to add a small method in one of our `Util` classes which would be called from our code base.

File: src/NewChain/Initialization.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as fs from 'fs';
 import * as path from 'path';
-import { Utils } from '@openst/mosaic.js';
+import { Utils as mosaicUtils } from '@openst/mosaic.js';
```
Comment (puneet-khushwani-eth) [inline src/NewChain/Initialization.ts]:
```suggestion
import { Utils as MosaicUtils } from '@openst/mosaic.js';
```

File: src/Utils.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Web3 = require('web3');
+
+export class Utils {
+
+  /**
+   * It provides checksum address using web3.
+   * @param address Address.
+   * @returns It returns checksum address.
+   */
+  public static toChecksumAddress(address: string): string {
```
Comment (puneet-khushwani-eth) [inline src/Utils.ts]: We should add unit tests for this method.

Comment (gulshanvasnani) [reply src/Utils.ts]: Added unit test case.

File: src/bin/mosaic-libraries.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ import Logger from '../Logger';
 import OriginChainInteract from '../NewChain/OriginChainInteract';
 import MosaicConfig from '../Config/MosaicConfig';
 import PublishMosaicConfig from '../Config/PublishMosaicConfig';
+import { Utils } from '../Utils';
```
Comment (sarvesh-ost) [inline src/bin/mosaic-libraries.ts]: Utils is exported as default. 
```suggestion
import Utils from '../Utils';
```

File: src/bin/mosaic-libraries.ts [hunk 5]
```diff
@@ -31,9 +32,9 @@ mosaic.action(
       const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(chain);
 
       mosaicConfig.originChain.chain = chain;
-      mosaicConfig.originChain.contractAddresses.gatewayLibAddress = gatewayLib.address;
-      mosaicConfig.originChain.contractAddresses.messageBusAddress = messageBus.address;
-      mosaicConfig.originChain.contractAddresses.merklePatriciaLibAddress = merklePatriciaProof.address;
+      mosaicConfig.originChain.contractAddresses.gatewayLibAddress = Utils.toChecksumAddress(gatewayLib.address);
```
Comment (sarvesh-ost) [inline src/bin/mosaic-libraries.ts]: Linter errors, break into two lines.

File: tests/Utils/toChecksumAddress.test.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import assert from '../test_utils/assert'
+import SpyAssert from '../test_utils/SpyAssert';
+import Utils from '../../src/Utils'
+
+describe('Utils.toChecksumAddress()', () => {
+  it('should be successfully', () => {
+    let address = '0x123Ad';
+
+    const web3Spy = sinon.stub(Web3.utils, 'toChecksumAddress').returns(address);
```
Comment (sarvesh-ost) [inline tests/Utils/toChecksumAddress.test.ts]: It's a pure function call, Do we need stub?

Comment (gulshanvasnani) [reply tests/Utils/toChecksumAddress.test.ts]: It is unit test and it has external dependency of `web3`
So i think we should stub the external dependencies.

**Pull request merged**